pkg/cli/options.go

```diff
@@ -139,3 +150,150 @@ func WithCompletion() Option {
 		return nil
 	}
 }
+
+// parseExternalPluginArgs parses external plugin arguments and trims plugins flag and its values from args.
+func parseExternalPluginArgs() (extPlugins []string, args []string) {
+	var externalPlugins string
+	args = make([]string, len(os.Args)-1)
+	copy(args, os.Args[1:])
+
+	// parse os.Args and pass everything to plugin except program name,
+	// --plugins flag and its value.
+	for i, item := range args {
+		if item == "--plugins" {
+			externalPlugins = args[i+1]
+			args = append(args[:i], args[i+2:]...)
+		}
+	}
+
+	extPlugins = strings.Split(externalPlugins, ".")
```

Comment:
Why split by `.`?

Reply:
Initially intended to trim the string before the dot and compare it to the plugin executable found, but didn't end up needing `extPlugins`. I am returning it in this function but didn't end up using this [here](https://github.com/kubernetes-sigs/kubebuilder/pull/2338/files#diff-1851a71300b8a592080399ec5bc32857e772b473fbc69b5fcec115870207d2f0R266).